### PR TITLE
Run package tests using make e2e-test

### PR DESCRIPTION
**Pull request by ShwethaKumbla**
## What this PR does / why we need it

This PR handles running addons package testing by triggering `make e2e-test` inside addons/packages/{packagename}/{version}/test folder.
## Details for the Release Notes (PLEASE PROVIDE)

```
Automation framework to trigger addons package testing by running `make e2e-test`

```

## Which issue(s) this PR fixes

Fixes: none

## Describe testing done for PR

Run `ginkgo -v -- --kubeconfig="$KUBE_CONFIG" --packages="external-dns" --version="0.8.0" --provider="docker" --cluster-type="standalone" --guest-cluster-name="tce-cluster-1" --create-cluster --tce-version="{vx.x.x}"`

The above command will create a standalone cluster and runs external-dns test by triggering `make e2e-test` command. 
## Special notes for your reviewer

NA

### CODE REVIEW

**Review by christianang — DISMISSED** [review]
LGTM

**Review by navidshaikh — CHANGES_REQUESTED** [review]
Thanks for making this change!

Added a comment inline

**Review by stmcginnis — COMMENTED** [review]
Looks good to me and I think comments have now been addressed. There are some merge conflicts with other modified files that need to be resolved now though.

While updating, please squash commits to have a single good commit message.

File: addons/packages/external-dns/0.8.0/test/Makefile
Comment (christianang) [inline addons/packages/external-dns/0.8.0/test/Makefile:28]: This will probably be resolved when you fix conflicts, but we changed this to `CGO_ENABLED=0 go run github.com/onsi/ginkgo/ginkgo -v e2e`. This has a couple advantages, one you don't need ginkgo downloaded and two it will use the version of ginkgo that is in the go.mod (not that I've had any particular issues with using different versions of ginkgo).

**Review by christianang — DISMISSED** [review]

**Review by navidshaikh — APPROVED** [review]

**Comment by ShwethaKumbla** [comment]
@christianang can you provide LGTM. This PR needs your approval again 🙂

**Comment by aman556** [comment]
I am also waiting for this one to get merge 😅

**Review by christianang — APPROVED** [review]

**Pull request merged**
